elastic/kibana
# [Connectors] Checking for undefined config and secrets during connector validation

Resolves https://github.com/elastic/kibana/issues/122346

## Summary

With [this PR](https://github.com/elastic/kibana/pull/116079), we added validation of connector config and secrets together. This validation was only added to the email connector. We've found that in cases where the connector `secrets` field is undefined, this will lead to a null accessor error (`Cannot read properties of undefined`) when executing connector action (either via the test connector UI or during actual action execution. The `secrets` value can become undefined due to decryption errors during migration.

## To Verify

1. Create an email connector in an older version of Kibana. Verify that you can test execution with this connector and it is successful.
2. On this branch, configure a different encryption key than before and start ES & Kibana using the data from 1)
3. Navigate to Connector UI to test execution of this connector. You should receive an execution error that says `secrets must be defined` instead of `Cannot read properties of undefined`

### Checklist
